### ISSUE TITLE
Double attribute penalty on item

### ISSUE THREAD

**Issue by LordOHelmet** [crnormand/gurps#770]
The heading doesn't say anything at all - I know!
But this one is hard to discribe. Best you try it for yourself and see what happens.

Here is the recipe:
1. Create a new Mook without skills and attacks

![mook](https://user-images.githubusercontent.com/80178910/121779963-7ac60080-cb9e-11eb-8aa2-f3cd903c3f0a.JPG)

2. Create two items. Attack and Attribute Penalty

![2items](https://user-images.githubusercontent.com/80178910/121780076-017add80-cb9f-11eb-8fe6-c3bc34fab9c2.JPG)

![attack](https://user-images.githubusercontent.com/80178910/121780167-73ebbd80-cb9f-11eb-9158-4877ffdee9e9.JPG)

![penalty](https://user-images.githubusercontent.com/80178910/121780170-7817db00-cb9f-11eb-98cd-1a0627da9c1b.JPG)

3. Drag both items on the Mook

![both equipt](https://user-images.githubusercontent.com/80178910/121780234-bf05d080-cb9f-11eb-92f7-3f4c020b801d.JPG)

4, Now unequip both.

![unequip](https://user-images.githubusercontent.com/80178910/121780330-0f7d2e00-cba0-11eb-858a-a0a835a1840a.JPG)

5. Finally reequip them in different orders. Attack first than Attribute Penalty; Attribute Penalty first than Attack. And see what happens to the Melee Weapon Attack Level

![level4](https://user-images.githubusercontent.com/80178910/121780450-a0540980-cba0-11eb-88af-d16e78b5841d.JPG)

![level5](https://user-images.githubusercontent.com/80178910/121780455-a4802700-cba0-11eb-86d4-9e2d21f6a91b.JPG)

Depending on the sequence used, the Attack Level differs.
When the "Attribute Penalty" is equipt first and "Attack" second - the attribute penalty is applied twice.

**Comment by crnormand** [comment]
Again, thank you for the detailed error report.   I believe I know the problem... I am applying the bonuses in serial fashion... and I should be applying the bonuses by type.

Meaning, I should apply all Skill bonuses first, then all Melee, etc.     It will mean a bit of rewriting, but this should be too hard to fix.

**Comment by crnormand** [comment]
Ok, I am pretty certain I got this correct this time... no seriously... stop laughing...  it could be true.    At first, I thought it mattered in which order I was building up the bonuses... and I tried to rearrange them to make it work.   but that wasn't the problem.

I wasn't thinking about the bonus situation correctly.  All bonuses should be applied first, and THEN we "collapse the quantum field" (as I like to call it), and calculate the OtF formulas in items.     

Now the order of the item doesn't matter because we build up all of the bonuses to attributes and skills first, from all of the items... and then we go through each items and calculate the features skill levels for that item.    At this point, the attribute values and skill levels are not changing, so it doesn't matter the order of the Items.

https://github.com/crnormand/gurps/pull/779
